mkosxinstallusb.sh

```diff
@@ -24,6 +24,11 @@ cleanup () {
     rm -f InstallESD.img BaseSystem.img
 }
 
+map_partitions () {
+  set +e
+  kpartx -sav "$1" | sed -r 's/^add\s+map\s+(\S+)p[0-9]+\s.*$/\/dev\/mapper\/\1/;t;d' | head -n1
```

Comment:
When this is called for the USB device in the line 129,

> stick_partitions="$(map_partitions "$stick_dev")"

then `kpartx` responds with the USB device name, like this:

> add map sdc1 (252:3): 0 63010783 linear 8:32 2048

which does not contain '/dev/loop?p?' and thus the regexp `(\S+)p[0-9]` does not match.

Due to that `$stick_partitions` in the line 129 becomes an empty string, so `mkfs.hfsplus` in the line 130 and `mount` in the line 138 fail. But the script does not stop because `map_partitions()` had `+e` set, so it copies 5Gb+ of files into the `/mnt/OSX_installer_drive`.

Reply:
The point itself is very valid. It is incorrect to hardcode a specific loop device because to the moment of the script run any number of them can be allocated.

Reply:
Ok. It looks like I just have to remove the `p` on line 24. Perhaps the capture group should also be changed to `[a-z]+` to be safe. (I forget the exact rules on regex greediness).

And I can't for the life of me remember why I put the `set +e`. I think I copied it from the `cleanup` function. There also isn't even a `set -e` in the first place (though I agree there should be) which makes this even more dumb. I'll take it out and add one to the start of the script.